zerver/views/message_fetch.py

```diff
@@ -870,10 +870,8 @@ def get_messages_backend(request: HttpRequest,
         if not is_web_public_compatible(narrow):
             return json_unauthorized()
 
-        realm = get_realm_from_request(request)
-        if realm is None:
-            return json_error(_("Invalid subdomain."))
-
+        realm = get_realm_from_request(request, True)
```

Comment:
Let's spell this as `get_realm_from_request(request, require_valid=True)`

And we can use https://zulip.readthedocs.io/en/latest/testing/mypy.html#using-overload-to-accurately-describe-variations to make that `assert realm is not None` implied by having passed that option.

Reply:
In general, a separate function is preferable to a mypy overload, since mypy does not check function bodies against their overloaded types.

Reply:
Done.